velox/connectors/hive/HiveDataSink.cpp

```diff
@@ -844,6 +822,32 @@ void HiveDataSink::splitInputRowsAndEnsureWriters() {
   }
 }
 
+// Returns the column indices of non-partition data columns.
+std::vector<column_index_t> HiveDataSink::getDataChannels(
+    const std::vector<column_index_t>& partitionChannels,
+    const column_index_t childrenSize) const {
+  std::vector<column_index_t> dataChannels;
+  dataChannels.reserve(childrenSize - partitionChannels.size());
+
+  for (column_index_t i = 0; i < childrenSize; i++) {
+    if (std::find(partitionChannels.cbegin(), partitionChannels.cend(), i) ==
+        partitionChannels.cend()) {
+      dataChannels.push_back(i);
+    }
+  }
+
+  return dataChannels;
+}
+
+std::string HiveDataSink::makePartitionDirectory(
```

Comment:
I had to make this function virtual because the write path for an iceberg table differs due to the presence of the /data directory. Since it became a member function, I also had to move it out of the namespace defined at line 40. Please let me know if there is a better way to achieve this.